Using default `tag_regex` raises DeprecationWarning

As of 10.1.2, accepting `scm-setuptools` configuration defaults and invoking `setuptools.get_version()` results in a DeprecationWarning:
```
../.tox/py314/lib/python3.14/site-packages/travertino/__init__.py:9: in _package_version
    return get_version(root="../../..", relative_to=file)  # pragma: no cover
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../.tox/py314/lib/python3.14/site-packages/setuptools_scm/_get_version.py:59: in get_version
    return _get_version_public(
../.tox/py314/lib/python3.14/site-packages/vcs_versioning/_get_version_impl.py:280: in get_version
    config = _config.Configuration(
../.tox/py314/lib/python3.14/site-packages/vcs_versioning/_config.py:334: in __post_init__
    warnings.warn(
E   DeprecationWarning: Configuration field 'tag_regex' is deprecated. Use 'tag.regex' instead.
```

This appears to be because the default argument of `tag_regex=_config.DEFAULT_TAG_REGEX` provided in the argument list of `get_version()` isn't identified as a default when it is processed by `vcs_versioning`.

As a workaround, passing in an explicit `tag_regex=None` argument seems to work. I'm not sure if using the same default in `get_version()` itself might be a fix, or have other consequences.

This could be related to #1429 and #1422.

--- a/vcs-versioning/changelog.d/1434.bugfix.md
+++ b/vcs-versioning/changelog.d/1434.bugfix.md
@@ -1,0 +1,1 @@
+Fix spurious ``DeprecationWarning`` for ``tag_regex`` when using default value via ``get_version()``.

--- a/vcs-versioning/src/vcs_versioning/_config.py
+++ b/vcs-versioning/src/vcs_versioning/_config.py
@@ -342,7 +342,6 @@ class Configuration:
                         "Cannot specify both 'tag_regex' (deprecated) and "
                         "'tag.regex'. Please use only 'tag.regex'."
                     )
-                # Replace with a new TagConfiguration to avoid mutating shared objects
                 self.tag = dataclasses.replace(
                     self.tag, regex=_check_tag_regex(tag_regex)
                 )

--- a/vcs-versioning/src/vcs_versioning/_get_version_impl.py
+++ b/vcs-versioning/src/vcs_versioning/_get_version_impl.py
@@ -10,7 +10,7 @@ from typing import Any, NoReturn
 
 from . import _config
 from . import _types as _t
-from ._config import Configuration
+from ._config import Configuration, TagConfiguration
 from ._environment import resolve_runtime_env
 
 # Backward-compat re-export used by vcs-versioning/setup.py
@@ -270,7 +270,18 @@ def get_version(
 
     version_cls = _validate_version_cls(version_cls, normalize)
     del normalize
-    tag_regex = parse_tag_regex(tag_regex)
+
+    if tag_regex is _config.DEFAULT_TAG_REGEX:
+        tag_config = TagConfiguration()
+    else:
+        warnings.warn(
+            "get_version() parameter 'tag_regex' is deprecated. "
+            "Use 'tag.regex' in pyproject.toml or pass a TagConfiguration "
+            "via Configuration(tag=...) instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        tag_config = TagConfiguration(regex=parse_tag_regex(tag_regex))
 
     scm_config = _config.ScmConfiguration.from_data(data=scm)
 
@@ -286,7 +297,6 @@ def get_version(
         version_file=version_file,
         version_file_template=version_file_template,
         relative_to=relative_to,
-        tag_regex=tag_regex,
         parentdir_prefix_version=parentdir_prefix_version,
         fallback_version=fallback_version,
         fallback_root=fallback_root,
@@ -296,6 +306,7 @@ def get_version(
         version_cls=version_cls,
         search_parent_directories=search_parent_directories,
         scm=scm_config,
+        tag=tag_config,
         _env=_env,
     )
     maybe_version = _get_version(config, force_write_version_files=True)

--- a/vcs-versioning/testing_vcs/test_config.py
+++ b/vcs-versioning/testing_vcs/test_config.py
@@ -7,7 +7,7 @@ import warnings
 
 import pytest
 from vcs_versioning import Configuration
-from vcs_versioning._config import TagConfiguration
+from vcs_versioning._config import DEFAULT_TAG_REGEX, TagConfiguration
 
 
 @pytest.mark.parametrize(
@@ -58,6 +58,21 @@ def test_deprecated_tag_regex_init_var() -> None:
     with pytest.warns(DeprecationWarning, match="Use 'tag.regex' instead"):
         conf = Configuration(tag_regex=tag_regex)
     assert conf.tag.regex is tag_regex
+
+
+@pytest.mark.issue(1434)
+def test_omitting_tag_regex_no_deprecation_warning() -> None:
+    """Omitting tag_regex (None default) should not emit DeprecationWarning."""
+    with warnings.catch_warnings():
+        warnings.simplefilter("error", DeprecationWarning)
+        Configuration()
+
+
+@pytest.mark.issue(1434)
+def test_explicit_default_tag_regex_warns() -> None:
+    """Explicitly passing DEFAULT_TAG_REGEX still warns (deprecated param)."""
+    with pytest.warns(DeprecationWarning, match="Use 'tag.regex' instead"):
+        Configuration(tag_regex=DEFAULT_TAG_REGEX)
 
 
 def test_tag_regex_conflict() -> None:

